app/scripts/modules/core/loadBalancer/loadBalancer.write.service.js

```diff
@@ -32,7 +32,7 @@ module.exports = angular
     function upsertLoadBalancer(loadBalancer, application, descriptor, params = {}) {
       var name = loadBalancer.clusterName || loadBalancer.name;
       loadBalancer.cloudProvider = loadBalancer.provider;
-      if (loadBalancer.healthCheckProtocol.indexOf('HTTP') === 0) {
+      if (loadBalancer.healthCheckProtocol != null && loadBalancer.healthCheckProtocol.indexOf('HTTP') === 0) {
```

Comment:
I intentionally wanted to catch `undefined` and `null`. WDYT @anotherchrisberry?